### PR TITLE
Allows to convert Spatie enum names to snake upper case

### PR DESCRIPTION
**Pull request by JoolsMcFly**
Hi.

This PR addresses issue https://github.com/rectorphp/rector/issues/8364 and lets you convert Spatie enum names to upper snake case.

### CODE REVIEW

**Comment by JoolsMcFly** [comment]
Could you tell me what the best way to test the rule with `toUpperSnakeCase` configuration set to `true` would be pls?

If I understand correctly naming convention makes it so that we can have only one test class for `SpatieEnumClassToEnumRector` and that is `SpatieEnumClassToEnumRectorTest`.
And since tests have only one configuration file loaded via `provideConfigFilePath` I have to pick one value of `toUpperSnakeCase` to test.

Ideas?

Thanks!

File: rules-tests/Php81/Rector/Class_/SpatieEnumClassToEnumRector/Fixture/some_class.php.inc [hunk 1]
```diff
@@ -26,6 +27,7 @@ enum StatusEnum : string
     case DRAFT = 'draft';
     case PUBLISHED = 'published';
     case ARCHIVED = 'archived';
+    case NEEDSREVIEW = 'needsReview';
```
Comment (TomasVotruba) [inline rules-tests/Php81/Rector/Class_/SpatieEnumClassToEnumRector/Fixture/some_class.php.inc]: Could you add this as a new test fixture? So we have it separated and linked to specific issue.

**Comment by TomasVotruba** [comment]
Only "ready" PRs trigger the CI, not drafts. So I took the liberty and triggered CI so we have more feedback

**Comment by TomasVotruba** [comment]
Could you also rebase on the `main`? There are some conflicts now

**Comment by JoolsMcFly** [comment]
Alright, merge conflicts are fixed and I moved upper case names  to a new test fixture.

Original tests are green so BC is ensured. 
What is the preferred way of adding a test that sets `toUpperSnakeCase` to true?

ps:
When I run `composer complete-check` it reports cs errors on files I haven't touched. I ran checks on an up-to-date `main` branch and there are errors there too. Should they be fixed in a separate PR?

File: rules/Php81/NodeFactory/EnumFactory.php [hunk 2]
```diff
@@ -117,12 +123,23 @@ private function createEnumCaseFromConst(ClassConst $classConst): EnumCase
...
     ): EnumCase {
         /** @var MethodTagValueNode $nodeValue */
         $nodeValue = $phpDocTagNode->value;
         $enumValue = $mapping[$nodeValue->methodName] ?? $nodeValue->methodName;
-        $enumName = strtoupper($nodeValue->methodName);
+        if ($enumNameInSnakeCase) {
+            $enumName = strtoupper(
+                Strings::replace(
+                    $nodeValue->methodName,
+                    '/(?<=[A-Z])(?=[A-Z][a-z])|(?<=[^A-Z])(?=[A-Z])|(?<=[A-Za-z])(?=[^A-Za-z])/',
```
Comment (samsonasik) [inline rules/Php81/NodeFactory/EnumFactory.php]: could you move it to constant and add `@see https://regex101.com/` uri to show the match, like other usage, eg:

https://github.com/rectorphp/rector-src/blob/a8cc19a3c0387f082fd432e04afce363a90ba5b0/rules/CodeQuality/Rector/Concat/JoinStringConcatRector.php#L26-L31

Comment (JoolsMcFly) [reply rules/Php81/NodeFactory/EnumFactory.php]: Hi.

Done!

Could not find on regex101.com how to add unit tests with substitution so I added three links. Is that OK?

The stackoverflow link includes a great explanation of the three parts of the regex too.

**Comment by TomasVotruba** [comment]
Looks good, thank you 👍

**Pull request merged**